ntoskrnl/ex/sysinfo.c

```diff
@@ -1350,21 +1353,14 @@ QSI_DEF(SystemHandleInformation)
             if ((HandleTableEntry->Object) &&
                 (HandleTableEntry->NextFreeTableEntry != -2))
             {
-                /* Increase of count of handles */
-                ++HandleInformation->NumberOfHandles;
```

Comment:
Are you sure this gets incremented only if `ExpLockHandleTableEntry` succeeds? (you're now using the `Index` value to hold the number)

Reply:
See Jira issue:
I am not 100% sure, as I do not know how to test this very case.
Nonetheless, it feels more self-consistent, and I assume current winetest was not reported as (intermittently) failing.

Reply:
When it comes to kernel development, you're either sure of yourself of what you're doing or you aren't. Assumptions are to be avoided.

If you're unsure then it's at best not worth touching the parts of the kernel you're clearly not fond with.